polyglot/piranha/src/models/unit_tests/source_code_unit_test.rs

```diff
@@ -1,3 +1,13 @@
+use std::{
```

Comment:
Going to assume that this placement of unit tests (rather than under `polyglot/piranha/src/test` is idiomatic for Rust, but still noting that it's a bit weird to me coming from either our Java or Go codebases.

Reply:
Rust's module system is very flexible and weird.  I too am figuring out the best organization strategy. This is something that  @mkr-plse  and I had agreed upon for the time being. 

 So this is how I have designed my tests: 
(i) Integration tests  (End to End tests)  They are as one would expect in `polyglot/piranha/src/tests`. 
([ref](https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html))
They basically test functionality of end to end refactorings, 
(ii) Unit tests: Function level tests.
 In`Rust` one to writes unit test within the same module where the logic is implemented ([Ref](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html)). In this way we can write tests for private methods too . Now adding tests, within the same file was too much . So for example [here](https://sourcegraph.com/github.com/uber/piranha/-/blob/polyglot/piranha/src/models/source_code_unit.rs?L129) I have annotated and added the "unit-test" sub module `souce_code_unit_test.rs` in `source_code_unit.rs`. 
Since I can basically give a `path` to the submodule (which is checked at compile time) I had liberty of putting the tests where-ever.  So i collected all the unit-tests for `models` and put them in one directory. 

Other options we had thought of : 


(i) Convert each `.rs.` file for which I want a unit test to a folder like : 
`source_code_unit`
       | - `mod.rs` (implements the logic)
       | - `test.rs` (implements the test)

(ii)  have a directory `src/unit-tests`.